From aedc22529c52c9f25236e3eedf2aec20dddf8a56 Mon Sep 17 00:00:00 2001
From: Martin Milata <mmilata@redhat.com>
Date: Thu, 4 Dec 2014 15:20:34 +0100
Subject: [PATCH] Only analyze vulnerabilities when coredump present

Also only generate core_backtrace if it's not already present.

Related to #882.

Signed-off-by: Martin Milata <mmilata@redhat.com>
---
 src/plugins/ccpp_event.conf | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/plugins/ccpp_event.conf b/src/plugins/ccpp_event.conf
index 62ff08a..f4f3828 100644
--- a/src/plugins/ccpp_event.conf
+++ b/src/plugins/ccpp_event.conf
@@ -16,9 +16,9 @@ EVENT=post-create analyzer=CCpp
         fi
         # Try generating backtrace, if it fails we can still use
         # the hash generated by abrt-action-analyze-c
-        ##satyr migration:
-        #satyr abrt-create-core-stacktrace "$DUMP_DIR"
-        abrt-action-generate-core-backtrace
+        [ ! -e core_backtrace ] && abrt-action-generate-core-backtrace
+        # Run GDB plugin to see if crash looks exploitable
+        [ -r coredump ] && abrt-action-analyze-vulnerability
         # Generate hash
         abrt-action-analyze-c &&
         abrt-action-list-dsos -m maps -o dso_list &&
-- 
2.4.3

